src/Purebred/Types/Addressbook.hs

```diff
@@ -0,0 +1,73 @@
+-- This file is part of purebred
+-- Copyright (C) 2022 Róman Joost
+--
+-- purebred is free software: you can redistribute it and/or modify
+-- it under the terms of the GNU Affero General Public License as published by
+-- the Free Software Foundation, either version 3 of the License, or
+-- (at your option) any later version.
+--
+-- This program is distributed in the hope that it will be useful,
+-- but WITHOUT ANY WARRANTY; without even the implied warranty of
+-- MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+-- GNU Affero General Public License for more details.
+--
+-- You should have received a copy of the GNU Affero General Public License
+-- along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+
+{- |
+
+Types for handling the retrieval of addresses
+
+-}
+module Purebred.Types.Addressbook
+  (
+    AddressRetriever(..)
+  , AddressRetrieverConfig(..)
+  , arRetrieve
+  , arConfig
+  , MuttAlias(..)
+  , maNick
+  , maAddress
+  ) where
+
+import Control.Lens (Lens', lens)
+import qualified Data.Text as T
+import GHC.Generics (Generic)
+import Control.DeepSeq (NFData(rnf), force)
+
+import Data.IMF (Address)
+
+data AddressRetrieverConfig = MuttAliasConfig FilePath | LDAPConfig String
+  deriving (Generic, NFData)
+
+data AddressRetriever = AddressRetriever
+  { _arRetrieve :: AddressRetrieverConfig -> T.Text -> IO [Address],
+    _arConfig :: AddressRetrieverConfig
+  }
+  deriving (Generic)
+
+instance NFData AddressRetriever where
+  rnf (AddressRetriever retrieve cfg) = AddressRetriever retrieve (force cfg) `seq` ()
+
+arRetrieve :: Lens' AddressRetriever (AddressRetrieverConfig -> T.Text -> IO [Address])
+arRetrieve = lens _arRetrieve (\r x -> r { _arRetrieve = x })
+
+arConfig :: Lens' AddressRetriever AddressRetrieverConfig
+arConfig = lens _arConfig (\r x -> r { _arConfig = x })
+
+data MuttAlias = MuttAlias
+  { _maNick :: T.Text,
+    _maAddress :: Address
+  } deriving (Show, Eq)
+
+maNick :: Lens' MuttAlias T.Text
+maNick = lens _maNick (\m x -> m { _maNick = x })
+
+maAddress :: Lens' MuttAlias Address
+maAddress = lens _maAddress (\m x -> m { _maAddress = x })
```

Comment:
The Mutt-specific stuff should be extracted to a separate file - as should each specific address book implementation.  e.g. `Purebred.AddressBook.MuttAliasFile`

Reply:
No worries. Moved.